src/liblzma/check/crc32_loongarch.h

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: 0BSD
+
+///////////////////////////////////////////////////////////////////////////////
+//
+/// \file       crc32_loongarch.h
+/// \brief      CRC32 calculation with LoongArch optimization
+//
+//  Authors:    Xi Ruoyao
+//
+///////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef LZMA_CRC32_LOONGARCH_H
+#define LZMA_CRC32_LOONGARCH_H
+
+#include <larchintrin.h>
+#include <sys/auxv.h>
+
+// Defined in section 9.1 of
+// "Software Development and Build Convention for LoongArch Architectures"
+// (https://github.com/loongson/la-softdev-convention/).  Hard code it
+// instead of including <bits/hwcap.h> for compatibility with Glibc <= 2.37.
+#define HWCAP_LOONGARCH_CRC32 (1 << 6)
+
+#if __loongarch_grlen >= 64
+#  define UNIT_SIZE 8
+#  define CRC_UNIT(buf, crc) __crc_w_d_w((int64_t)aligned_read64le(buf), crc)
+#else
+#  define UNIT_SIZE 4
+#  define CRC_UNIT(buf, crc) __crc_w_w_w((int32_t)aligned_read32le(buf), crc)
+#endif
+
+static uint32_t
```

Comment:
Extra newline needed before this line for code style.

Reply:
Done.